### DISCUSSION TITLE
Plugins array in docusaurus.Config.js

### DISCUSSION THREAD

**Discussion by Karan-S-dev** [facebook/docusaurus#8005] [Q&A] [answered]
I am using docusaurus to built a website which will show the documentation of multiple products. Like more than 50 products.
So my plugins array will look like this,

plugins: [
    [
      "@docusaurus/plugin-content-docs",
      {
        id: "product1",
        path: "product1",
        routeBasePath: "product1",
        sidebarPath: require.resolve("./sidebars.js"),
      },
    ],
    [
      "@docusaurus/plugin-content-docs",
      {
        id: "product2",
        path: "product2",
        routeBasePath: "product2",
        sidebarPath: require.resolve("./sidebars.js"),
 
      },
    ],
    [
      "@docusaurus/plugin-content-docs",
      {
        id: "product3",
        path: "product3",
        routeBasePath: "product3",
        sidebarPath: require.resolve("./sidebars.js"),
      },
    ],
    [
      "@docusaurus/plugin-content-docs",
      {
        id: "product4",
        path: "product4",
        routeBasePath: "product4",
        sidebarPath: require.resolve("./sidebars.js"),
      },
    ],
],

My doubt is what is the need of this sidebarPath thing in every structure. Is it necessary to add this is every structure. What will happen if I remove this sidebarPath line from every structure?

**Comment by slorber** [comment, marked as answer]
`sidebarPath: undefined` would lead to an auto-generated sidebar

See https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-content-docs#sidebarPath

![CleanShot 2022-08-25 at 13 18 59@2x](https://user-images.githubusercontent.com/749374/186651004-29fdb524-fcfb-4536-8176-d5dda6ca1f3c.png)

--- 

If you use versioning, you may be interested by this issue: https://github.com/facebook/docusaurus/issues/7968


---

If you have a large set of documentation for each product, you might want to create one standalone site per product instead of gathering all products into a single doc. You can share infrastructure (custom theming, shared config on multiple sites)

See for example Figma using 2 different Docusaurus sites:
- https://www.figma.com/plugin-docs/
- https://www.figma.com/widget-docs/

Another example from Palo Alto:
- https://xsoar.pan.dev/
- https://prisma.pan.dev/

**Reply by Karan-S-dev** [reply]
In that sidebar field, if i give false sidebars are not coming. But the sidebars are coming and working perfectly. when i give undefined.
So if i proceed  with undefined value, it will not give any impedance in the future right?

Regarding that infrastructure, i will defenitely look into that.

**Reply by slorber** [reply]
The official doc says:

> Path to sidebar configuration. Use false to disable sidebars, or undefined to create a fully autogenerated sidebar.



> So if i proceed with undefined value, it will not give any impedance in the future right?

I'm not sure what you mean, but if we change this behavior, it will be documented as a breaking change
